aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,853 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+        "vneg.f32 %[res0], %[in_0]\n\t"
+        "vneg.f32 %[res1], %[in_1]\n\t"
+        : [res0] "=w" (res0), [res1] "=w" (res1)
+        : [in_0] "w" (values.val[0]), [in_1] "w" (values.val[1])
+        :);
+    return Vec256<float>(res0, res1);
+  }
+  Vec256<float> round() const {
+    // We do not use std::round because we would like to round midway numbers to the nearest even integer.
+    return map(at::native::round_impl);
+  }
+  Vec256<float> tan() const {
+    return map(std::tan);
+  }
+  Vec256<float> tanh() const {
+    return map(std::tanh);
+  }
+  Vec256<float> trunc() const {
+    int32x4_t int_res0, int_res1;
+    float32x4_t res0, res1;
+    __asm__ (
+        "vcvt.s32.f32 %[int_res0], %[in_0]\n\t"
+        "vcvt.s32.f32 %[int_res1], %[in_1]\n\t"
+        "vcvt.f32.s32 %[res0], %[int_res0]\n\t"
+        "vcvt.f32.s32 %[res1], %[int_res1]\n\t"
+        : [res0] "=w" (res0), [res1] "=w" (res1),
+          [int_res0] "+w" (int_res0), [int_res1] "+w" (int_res1)
+        : [in_0] "w" (values.val[0]), [in_1] "w" (values.val[1])
+        :);
+    return Vec256<float>(res0, res1);
+  }
+  Vec256<float> lgamma() const {
+    return map(std::lgamma);
+  }
+  Vec256<float> sqrt() const {
+    register float32x4_t res0 asm("q3");
+    register float32x4_t res1 asm("q4");
+    __asm__ (
+        "vmov.f32 %[res0], %[in_0]\n\t"
+        "vmov.f32 %[res1], %[in_1]\n\t"
+        "vsqrt.f32 s12, s12\n\t"
+        "vsqrt.f32 s13, s13\n\t"
+        "vsqrt.f32 s14, s14\n\t"
+        "vsqrt.f32 s15, s15\n\t"
+        "vsqrt.f32 s16, s16\n\t"
+        "vsqrt.f32 s17, s17\n\t"
+        "vsqrt.f32 s18, s18\n\t"
+        "vsqrt.f32 s19, s19\n\t"
+        : [res0] "+w" (res0), [res1] "+w" (res1)
+        : [in_0] "w" (values.val[0]), [in_1] "w" (values.val[1])
+        :);
+    return Vec256<float>(res0, res1);
+  }
+  Vec256<float> reciprocal() const {
+    float32x4_t res0, res1;
+    __asm__ (
+        "vrecpe.f32 %[res0], %[in_0]\n\t"
+        "vrecpe.f32 %[res1], %[in_1]\n\t"
+        : [res0] "=w" (res0), [res1] "=w" (res1)
+        : [in_0] "w" (values.val[0]), [in_1] "w" (values.val[1])
+        :);
+    return Vec256<float>(res0, res1);
+  }
+  Vec256<float> rsqrt() const {
+    float32x4_t res0, res1;
+    __asm__ (
+        "vrsqrte.f32 %[res0], %[in_0]\n\t"
+        "vrsqrte.f32 %[res1], %[in_1]\n\t"
+        : [res0] "=w" (res0), [res1] "=w" (res1)
+        : [in_0] "w" (values.val[0]), [in_1] "w" (values.val[1])
+        :);
+    return Vec256<float>(res0, res1);
+  }
+  Vec256<float> pow(const Vec256<float> &exp) const {
+    __at_align32__ float tmp[size()];
+    __at_align32__ float tmp_exp[size()];
+    store(tmp);
+    exp.store(tmp_exp);
+    for (int64_t i = 0; i < size(); i++) {
+      tmp[i] = std::pow(tmp[i], tmp_exp[i]);
+    }
+    return loadu(tmp);
+  }
+  Vec256<float> operator==(const Vec256<float>& other) const {
+    float32x4_t res0, res1;
+    __asm__ (
+        "vceq.f32 %[res0], %[in_0], %[other_in_0]\n\t"
```

Comment:
Does this modify the flags register.  I think not, but if it does you need to add "cc" to the clobber list.

Reply:
I will check. Thanks for this catch.